src/Ztm.WebApi/HttpResponseExceptionFilter.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Newtonsoft.Json;
+
+namespace Ztm.WebApi
+{
+    class ErrorDetails
+    {
+        public object Message { get; set; }
+
+        public override string ToString()
+        {
+            return JsonConvert.SerializeObject(this);
+        }
+    }
+
+    public class HttpResponseExceptionFilter : IActionFilter
+    {
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+        }
+
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+            if (context.Exception is HttpResponseException exception)
+            {
+                context.Result = new JsonResult(new ErrorDetails{Message = exception.Value})
```

Comment:
We should not use `JsonResult` due to it will break ASP.NET automatic response type.

Reply:
Create controller to handle error instead